stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/usecase/QueryMembersImpl.kt

```diff
@@ -0,0 +1,50 @@
+package io.getstream.chat.android.livedata.usecase
+
+import androidx.annotation.CheckResult
+import io.getstream.chat.android.client.api.models.QuerySort
+import io.getstream.chat.android.client.call.Call
+import io.getstream.chat.android.client.call.CoroutineCall
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.utils.FilterObject
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.livedata.ChatDomainImpl
+
+public interface QueryMembers {
+
+    @CheckResult
+    public operator fun invoke(
+        channelType: String,
+        channelId: String,
+        offset: Int = 0,
+        limit: Int = 0,
+        filter: FilterObject = FilterObject(),
+        sort: QuerySort<Member> = QuerySort<Member>().desc("created_at"),
+        members: List<Member> = emptyList(),
+    ): Call<List<Member>>
+}
+
+internal class QueryMembersImpl(private val domainImpl: ChatDomainImpl) : QueryMembers {
+    override fun invoke(
+        channelType: String,
+        channelId: String,
+        offset: Int,
+        limit: Int,
+        filter: FilterObject,
+        sort: QuerySort<Member>,
+        members: List<Member>,
+    ): Call<List<Member>> {
+        return CoroutineCall(domainImpl.scope) {
+            if (domainImpl.isOffline() && domainImpl.offlineEnabled) {
+                val offlineMembers: List<Member> =
+                    domainImpl.channel(channelType, channelId).members.value ?: emptyList()
+                Result(offlineMembers)
+            } else {
+                domainImpl
```

Comment:
Shouldn't we store members in the database and also add them to `ChannelController:members` and `Channel:members`?

Reply:
Since this function provides the ability to obtain a subset of the actual members, should we be overwriting the channel members in the db?

Reply:
I think for fresh data obtained from BE we need to update those values in local DB. For offline case we can fetch data from DB (like you said emulating BE behaviour)

Reply:
To be honest I'm not sure what we should do in that case. Maybe this function should return 'ChannelController' which exposes Members LiveData and post updates there (like we do with messages) instead of returning subset of members? I think it might be worth it to check how other SDKs deal with it (I'm not sure if other teams already implemented that) or we should discuss that with other team members?

Reply:
I'll check out the other SDKs and see how they handle it as well.

Reply:
IMO, fetching data from DB directly is better than exposing the LiveData value of `ChannelController.` Because:
1) It might be not updated
2) We want to have one source of truth in the offline SDK. We agreed to have DB as such source